### PR TITLE
Features

### PR DESCRIPTION
**Pull request by 4i8**
Support CommonJS And ES6 and Support 
Auto Extract the frames automatically and also support array of Buffers or url or a local path

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
-import crc from 'crc';
-
+const crc = require("crc");
```
Comment (TomasHubelbauer) [inline index.js:1]: Unfortunately, I don't want to support CJS. ESM is the future and as such should be the only used import style in new code IMO. If there is desire for CJS support, it is going to have to be in a fork or a Github Actions workflow can be introduced to transpile ESM to CJS and drop that as an artifact.

File: index.js [hunk 2]
```diff
@@ -10,90 +11,117 @@ import crc from 'crc';
...
-export default function makeApng(/** @type {Buffer[]} */ buffers, /** @type {number} */ delay) {
-  function findChunk(buffer, type, offset = 8) {
-    while (offset < buffer.length) {
-      const chunkLength = buffer.readUInt32BE(offset);
-      const chunkType = buffer.slice(offset + 4, offset + 8).toString('ascii');
+module.exports = function makeApng(
+  /** @type {Buffer[]} */ buffers,
+  /** @type {number} */ delay
+) {
+  return new Promise(async (resolve) => {
```
Comment (TomasHubelbauer) [inline index.js:18]: This internal Promise seems redundant, the function itself should be async.

File: index.js [hunk 3]
```diff
@@ -10,90 +11,117 @@ import crc from 'crc';
...
-    fctl.writeUInt32BE(0, 24); // Y offset
-    const { numerator, denominator } = delay(index);
-    fctl.writeUInt16BE(numerator, 28); // Frame delay - fraction numerator
-    fctl.writeUInt16BE(denominator, 30); // Frame delay - fraction denominator
-    fctl.writeUInt8(0, 32); // Dispose mode
-    fctl.writeUInt8(0, 33); // Blend mode
-    fctl.writeUInt32BE(crc.crc32(fctl.slice(4, 34)), 34); // CRC
+      const fctl = Buffer.alloc(38);
+      fctl.writeUInt32BE(26, 0); // Length of chunk
+      fctl.write("fcTL", 4); // Type of chunk
```
Comment (TomasHubelbauer) [inline index.js:73]: Please keep the existing formatting style: single quotes.

**Review by TomasHubelbauer — CHANGES_REQUESTED** [review]
I added a few notes on my phone, will do a full review later when I'm at the computer.

File: index.js (hunk 3 above)
Comment (4i8) [reply index.js:73]: > Please keep the existing formatting style: single quotes.

Extension added automatically

File: index.js (hunk 1 above)
Comment (4i8) [reply index.js:1]: Not problem

**Pull request closed without merging**